ArduPlane/Log.cpp

```diff
@@ -483,8 +483,12 @@ const struct LogStructure Plane::log_structure[] = {
       "SONR", "QffBf",   "TimeUS,Dist,Volt,Cnt,Corr" },
     { LOG_ARM_DISARM_MSG, sizeof(log_Arm_Disarm),
       "ARM", "QBH", "TimeUS,ArmState,ArmChecks" },
+    { LOG_THERMAL_MSG, sizeof(SoaringController::log_tuning),
+      "SOAR", "QfffffffLLfffI", "TimeUS,nettorate,dx,dy,x0,x1,x2,x3,lat,lng,alt,dx_w,dy_w" },
```

Comment:
I think the SOAR and VAR log messages are better done with the new Log_Write() methods.
See this example in AP_TECS:
https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_TECS/AP_TECS.cpp#L1058
that way you don't need the structure definition and the code is a lot clearer. It is a bit slower, but I don't think that will matter for these messages.